src/Model/HasLegacyAttributesTrait.php

```diff
@@ -53,6 +58,13 @@ trait HasLegacyAttributesTrait
      */
     public $mutable = false;
 
+    /**
+     * Attributes that are cast to another types when fetched from database.
+     *
+     * @var array
+     */
+    protected $casts = [];
```

Comment:
As a new property, I think you could set type.
```suggestion
    protected array $casts = [];
```

Reply:
Done: https://github.com/leroy-merlin-br/mongolid/pull/220/commits/ea1efb32835be3a0deba98888a58b39cbd8409d8